okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/RemotingOkHttpCall.java

```diff
@@ -274,13 +263,38 @@ public Void visit(QosException.Unavailable exception) {
                 } else {
                     log.debug("Rescheduling call after backoff",
                             SafeArg.of("backoffMillis", backoff.get().toMillis()));
-                    scheduleExecution(callback, backoff.get());
+                    redirectToUrl(
+                            callback,
+                            call,
+                            // Redirect to the "next" URL, whichever that may be, after backing off.
+                            () -> urls.redirectToNext(request().url()),
+                            "Failed to determine valid redirect URL for base URLs " + urls.getBaseUrls(),
+                            request -> scheduleExecution(() -> client.newCallWithMutableState(request,
+                                    backoffStrategy, maxNumRelocations - 1).enqueue(callback), backoff.get()));
                 }
                 return null;
             }
         };
     }
 
+    private void redirectToUrl(Callback callback, Call call, Supplier<Optional<HttpUrl>> optionalUrlSupplier,
```

Comment:
think this construction is pretty hard to grok, maybe be more readable with a little bit of code duplication in the visit() methods. summarizing:
- RetryOther should fail if maxNumRetries <= 0 or no URL can be determined
- Unavailable should fail if backoff is empty or no URL can be determined

Reply:
yeah realized i was possibly going overboard with the DRY there